File: ios/AlphaCityStampTour/AlphaCityStampTour/Network/DTO/AuthDTO.swift
//
//  AuthDTO.swift
//  AlphaCityStampTour
//

import Foundation

// MARK: - 요청

struct LoginRequest: Encodable {
    let email: String
    let password: String
}

struct RegisterRequest: Encodable {
    let email: String
    let password: String
    let nickname: String
}

struct SocialLoginRequest: Encodable {
    let provider: String
    let accessToken: String
}

struct RefreshRequest: Encodable {
    let refreshToken: String
}

struct ResetPasswordRequest: Encodable {
    let email: String
    let newPassword: String
}

struct SendCodeRequest: Encodable {
    let phone: String
}

struct VerifyCodeRequest: Encodable {
    let phone: String
    let code: String
}

// MARK: - 응답

struct AuthData: Decodable {
    let token: String
    let refreshToken: String
    let user: UserData
}

struct UserData: Decodable {
    let id: Int
    let email: String
    let nickname: String
    let profileImage: String?
}

struct TokenData: Decodable {
    let token: String
    let refreshToken: String
}

struct EmptyData: Decodable {}

File: android/app/src/main/java/com/alphacity/stamptour/network/dto/AuthDto.kt
package com.alphacity.stamptour.network.dto

import kotlinx.serialization.Serializable

// === 요청 ===

@Serializable
data class LoginRequest(
    val email: String,
    val password: String,
)

@Serializable
data class RegisterRequest(
    val email: String,
    val password: String,
    val nickname: String,
    val phone: String? = null,
    val name: String? = null,
)

@Serializable
data class SocialLoginRequest(
    val provider: String,
    val accessToken: String,
)

@Serializable
data class RefreshRequest(
    val refreshToken: String,
)

@Serializable
data class ResetPasswordRequest(
    val email: String,
    val newPassword: String,
)

@Serializable
data class SendCodeRequest(
    val phone: String,
)

@Serializable
data class VerifyCodeRequest(
    val phone: String,
    val code: String,
)

// === 응답 ===

@Serializable
data class AuthData(
    val token: String,
    val refreshToken: String,
    val user: UserData,
)

@Serializable
data class UserData(
    val id: Int,
    val email: String,
    val nickname: String,
    val profileImage: String? = null,
)

@Serializable
data class TokenData(
    val token: String,
    val refreshToken: String,
)
